### ISSUE TITLE
[BUG] `Enqueue` shouldn't call `schedule`

### ISSUE THREAD

**Issue by xuyang2** [hibiken/asynq#158]
**Describe the bug**
When you call `client.Enqueue(task)` it shouldn't be calling `schedule()`

**To Reproduce**
Steps to reproduce the behavior (Code snippets if applicable):
just call `client.Enqueue(task)`

**Expected behavior**
When you call `client.Enqueue(task)` it should call `enqueue()`

**Screenshots**
![image](https://user-images.githubusercontent.com/6643001/83714119-764a3c00-a65c-11ea-92a9-fd5ac0da2f47.png)

**Environment (please complete the following information):**
 - OS: Windows
 - Version of `asynq` package: a38f628f3b24090a63bececc44d87e11d9edbb75

**Additional context**
```go
t1 := time.Now()
t2 := time.Now()
fmt.Printf("delta nanos: %v\n", t2.UnixNano()-t1.UnixNano()) // Probably 0
fmt.Printf("t2 after t1: %v\n", t2.After(t1))                // Probably false
fmt.Printf("t2 not before t1: %v\n", !t2.Before(t1))         // Should be true
```
https://play.golang.org/p/K41KEKdMIQg

I guess L235 can change a little to make it more reliable. eg.
```go
if time.Now().UnixNano() >= t.UnixNano() {
if !time.Now().Before(t) {
if !t.After(time.Now()) {
```

https://github.com/hibiken/asynq/blob/a38f628f3b24090a63bececc44d87e11d9edbb75/client.go#L235-L239

**Comment by hibiken** [comment]
@xuyang2 thank you for opening this issue! 
I'll look into this once I'm finished with my current feature work. But if someone wants to take this bug, please feel free to take this bug and open a PR 👍

**Comment by hibiken** [comment]
@xuyang2 I had trouble reproducing this bug on my machine (the conditional always evaluated to `true` for me) so I couldn't add a test case to reproduce the bug.
But it should be fixed by #171. Please reopen this issue if you continue to see the behavior. thanks!
